wafer/schedule/views.py

```diff
@@ -203,3 +210,70 @@ def get_context_data(self, **kwargs):
                     # Must overlap with current slot
                     item['note'] = 'current'
         return context
+
+
+class ScheduleItemViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows groups to be viewed or edited.
+    """
+    queryset = ScheduleItem.objects.all()
+    serializer_class = ScheduleItemSerializer
+
+
+class ScheduleEditView(TemplateView):
+    template_name = 'wafer.schedule/edit_schedule.html'
+    date_field = 'date'
+    allow_future = True
+
```

Comment:
This view should be protected by appropriate admin permissions.

While the permissions on the rest API will prevent unauthorized changes, we don't want to leak unpublished information in the schedule carelessly.